File: Data Structures & Algorithms/string-encode-and-decode/submission-0.cpp
class Solution {
public:

    string encode(vector<string>& strs) {
        string final = "";
        for (string s: strs){
            final+=to_string(s.size());
            final+='#';
            final+=s;
        }
        return final;
    }

    vector<string> decode(string s) {
        vector<string> ans;
        int pos = 0;
        while (pos<s.size()){
            string num = "";
            string cur = "";
            while (s[pos]!='#'){
                num+=s[pos];
                pos++;
            }
            pos++;
            int n = stoi(num);
            for (int i=0; i<n; i++){
                cur+=s[pos+i];
            }
            pos+=n;
            ans.push_back(cur);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/decode-ways/submission-0.py
class Solution:
    def numDecodings(self, s: str) -> int:
        ans = [1]
        for i in range(len(s)):
            cur = 0
            if (s[i]!='0'):
                cur+=ans[-1]
            if (i>0 and int(s[i-1:i+1])>=10 and int(s[i-1:i+1])<=26):
                cur+=ans[-2]
            ans.append(cur)
        print(ans)
        return ans[-1]